dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -636,47 +629,33 @@ def _construct_collection_plan(cls, dataset_info):
-        # want to apply filters, aggregate files, or calculate
-        # divisions.
-        _need_aggregation_stats = chunksize or (
-            int(split_row_groups) > 1 and aggregation_depth
-        )
-        if len(index_cols) > 1:
-            gather_statistics = False
-        elif not _need_aggregation_stats and filters is None and len(index_cols) == 0:
-            gather_statistics = False
-
-        # Make sure gather_statistics allows filtering
-        # (if filters are desired)
-        if filters:
-            # Filters may require us to gather statistics
-            if gather_statistics is False and pf.cats:
-                warnings.warn(
-                    "Filtering with gather_statistics=False. "
-                    "Only partition columns will be filtered correctly."
-                )
-            elif gather_statistics is False:
-                raise ValueError("Cannot apply filters with gather_statistics=False")
-            elif not gather_statistics:
-                gather_statistics = True
-
         # Determine which columns need statistics.
+        # At this point, gather_statistics is only True if
+        # the user specified calculate_divisions=True
         filter_columns = {t[0] for t in flatten(filters or [], container=list)}
         stat_col_indices = {}
+        _index_cols = index_cols if (gather_statistics and len(index_cols) == 1) else []
         for i, name in enumerate(pf.columns):
-            if name in index_cols or name in filter_columns:
+            if name in _index_cols or name in filter_columns:
                 stat_col_indices[name] = i
 
-        # If the user has not specified `gather_statistics`,
-        # we will only do so if there are specific columns in
-        # need of statistics.
-        # NOTE: We cannot change `gather_statistics` from True
-        # to False (even if `stat_col_indices` is empty), in
-        # case a `chunksize` was specified, and the row-group
-        # statistics are needed for part aggregation.
-        if gather_statistics is None:
-            gather_statistics = bool(stat_col_indices)
+        # We now have enough info to set the final gather_statistics
+        gather_statistics = None if not gather_statistics else True
+        if (
+            chunksize
+            or (int(split_row_groups) > 1 and aggregation_depth)
+            or filter_columns.intersection(stat_col_indices)
+        ):
+            # Need to gather statistics if we are aggregating files
+            # or filtering
+            # NOTE: Should avoid gathering statistics when the agg
+            # does not depend on a row-group statistic
+            gather_statistics = True
+        elif not stat_col_indices or gather_statistics is None:
+            # Not aggregating files/row-groups.
+            # We only need to gather statistics if `stat_col_indices`
+            # is populated
+            gather_statistics = False
```

Comment:
Same here:

```suggestion
        # We now have enough info to set the final gather_statistics
        if (
            chunksize
            or (int(split_row_groups) > 1 and aggregation_depth)
            or filter_columns.intersection(stat_col_indices)
        ):
            # Need to gather statistics if we are aggregating files
            # or filtering
            # NOTE: Should avoid gathering statistics when the agg
            # does not depend on a row-group statistic
            gather_statistics = True
        elif not stat_col_indices or gather_statistics is None:
            # Not aggregating files/row-groups.
            # We only need to gather statistics if `stat_col_indices`
            # is populated
            gather_statistics = False
        else:
            gather_statistics = None if not gather_statistics else True
```

It's a shame all this logic is duplicated, perhaps we can figure out a way to simplify this in the future.

Reply:
>It's a shame all this logic is duplicated, perhaps we can figure out a way to simplify this in the future.

Agreed - I moved this logic into `parquet/utils.py`